### PR TITLE
Feature/coin metadata 1642

### PR DESCRIPTION
**Pull request by Brando753**

### CODE REVIEW

File: cmd/ebrelayer/txs/parser.go [hunk 1]
```diff
@@ -79,6 +79,9 @@ func EthereumEventToEthBridgeClaim(valAddr sdk.ValAddress, event types.EthereumE
 	witnessClaim.CosmosReceiver = recipient.String()
 	witnessClaim.Amount = amount
 	witnessClaim.ClaimType = event.ClaimType
+	witnessClaim.Decimals = event.Decimals
+	witnessClaim.TokenName = event.Name
+	witnessClaim.DenomHash = ethbridge.GetDenomHash(networkDescriptor, symbol, event.Decimals, event.Name, event.Symbol)
```
Comment (ElliotFriedman) [inline cmd/ebrelayer/txs/parser.go:84]: We decided to move away from this hashing denom model.

Comment (ElliotFriedman) [reply cmd/ebrelayer/txs/parser.go:84]: We can remove this hash.

File: proto/sifnode/ethbridge/v1/types.proto [hunk 2]
```diff
@@ -34,6 +34,14 @@ message EthBridgeClaim {
...
   ];
   ClaimType claim_type = 10;
+  // Name of coin different then symbol
+  string token_name = 11 [ (gogoproto.moretags) = "yaml:\"token_name\"" ];
+  // Number of decimals in ERC20 token
+  int32 decimals = 12 [ (gogoproto.moretags) = "yaml:\"token_decimals\"" ];
+  // The hashed concatination of ERC20 address, network descriptor
+  // Token decimals, name and symbol
+  // Marked optional because messages sent by the relayer may not be hashed
+  string denom_hash = 13 [ (gogoproto.moretags) = "yaml:\"denom_hash\"" ];
```
Comment (ElliotFriedman) [inline proto/sifnode/ethbridge/v1/types.proto:44]: Let's remove this.

File: x/ethbridge/types/claim.go [hunk 3]
```diff
@@ -16,30 +18,36 @@ func NewEthBridgeClaim(
...
-		TokenContractAddress:  tokenContact.String(),
+		TokenContractAddress:  tokenContract.String(),
 		EthereumSender:        ethereumSender.String(),
 		CosmosReceiver:        cosmosReceiver.String(),
 		ValidatorAddress:      validator.String(),
 		Amount:                amount,
 		ClaimType:             claimType,
+		TokenName:             tokenName,
+		Decimals:              decimals,
+		DenomHash:             denomHash,
```
Comment (ElliotFriedman) [inline x/ethbridge/types/claim.go:44]: same comment as above on the denom hash.

File: x/ethbridge/types/claim.go [hunk 4]
```diff
@@ -49,8 +57,58 @@ func (claim *EthBridgeClaim) GetProphecyID() []byte {
 		claim.CosmosReceiver,
 		claim.Amount.String(),
 		claim.ClaimType.String(),
+		claim.TokenName,
+		strconv.Itoa(int(claim.Decimals)),
+		claim.DenomHash,
```
Comment (ElliotFriedman) [inline x/ethbridge/types/claim.go:62]: Remove

File: x/ethbridge/types/claim.go [hunk 5]
```diff
@@ -49,8 +57,58 @@ func (claim *EthBridgeClaim) GetProphecyID() []byte {
...
 	)
 	claimBytes := []byte(allContentString)
 	hashBytes := crypto.Keccak256(claimBytes)
 	return hashBytes
 }
+
+func GetDenomHash(
```
Comment (ElliotFriedman) [inline x/ethbridge/types/claim.go:69]: remove

**Review by ElliotFriedman — CHANGES_REQUESTED** [review]

**Comment by Brando753** [comment]
Code merged into Pull Request #1671

**Pull request closed without merging**